service/src/routes/capture.ts

```diff
@@ -122,6 +122,36 @@ export default class CaptureRouter extends SelfAwareRouter {
 
       }));
 
+      this.router.put('/:id(\\d+)', check.validParams(schema.idParams), check.validBody(schema.putCaptureBody),
+            this.handleHttpErrors(async (request, response) => {
+
+         const capture = await captureDao.getCapture(request.params.id);
+         if (!capture || !capture!.envId) {
+            throw new HttpError(http.NOT_FOUND);
+         }
+
+         const captures = await captureDao.getCapturesForEnvironment(capture!.envId!);
+         if (!captures) {
+            throw new HttpError(http.INTERNAL_SERVER_ERROR);
+         }
+
+         let nameAvailable = true;
+
+         captures.forEach( (value, index, arr) => {
```

Comment:
A simpler way to do this would be to make a `getCaptureByName(name: string): Promise<ICapture | null>` function on the capture-dao, call it with the given name, and make sure that null is returned.